spec/03-plugins/37-opentelemetry/04-exporter_spec.lua

```diff
@@ -243,6 +243,80 @@ for _, strategy in helpers.each_strategy() do
       end)
     end
 
+
+    describe("With config.sampling_rate unset, using global sampling rate: 0.5", function ()
+      local mock
+      local sampling_rate = 0.5
+       -- this trace_id is always sampled with 0.5 rate
+      local sampled_trace_id = "92a54b3e1a7c4f2da9e44b8a6f3e1dab"
+       -- this trace_id is never sampled with 0.5 rate
+      local non_sampled_trace_id = "4bf92f3577b34da6a3ce929d0e0e4736"
+
+      lazy_setup(function()
+        bp, _ = assert(helpers.get_db_utils(strategy, {
+          "services",
+          "routes",
+          "plugins",
+        }, { "opentelemetry" }))
+
+        setup_instrumentations("all", {}, nil, nil, nil, nil, sampling_rate)
+        mock = helpers.http_mock(HTTP_SERVER_PORT, { timeout = HTTP_MOCK_TIMEOUT })
+      end)
+
+      lazy_teardown(function()
+        helpers.stop_kong()
+        if mock then
+          mock("close", true)
+        end
+      end)
+
+      it("does not sample spans when trace_id == non_sampled_trace_id", function()
+        local cli = helpers.proxy_client(7000, PROXY_PORT)
+        local r = assert(cli:send {
+          method  = "GET",
+          path    = "/",
+          headers = {
+            traceparent = "00-" .. non_sampled_trace_id .. "-0123456789abcdef-01"
+          }
+        })
+        assert.res_status(200, r)
+
+        cli:close()
+
+        ngx.sleep(2)
```

Comment:
Why do we sleep for 2 seconds? 🤔

Reply:
because (this is partially copy pasted from the test above) we have to wait long enough to be fairly certain the tracing backend (mock) would have received the tracing data (if any). Checking immediately would always return nothing, so in the test below, where we expect data, we use wait_until (could be actually updated to use `eventually`), but here we just wait an arbitrary amount of time to ensure that if there as any data, it would be there by now.

Reply:
Ok I see. Huh that's the perpetual problem of waiting for something **not** to happen 😅

Reply:
exactly :D

Reply:
That sounds like a place we should use `wait_until`. Is it possible?